cvrf2csaf/section_handlers/document_notes.py

```diff
@@ -3,11 +3,52 @@
 
 
 class DocumentNotes(SectionHandler):
-    def __init__(self):
-        super().__init__()
+    def __init__(self, config):
+        super().__init__(config=config)
+
+        self.enum_categories = {
+            'description',
+            'details',
+            'faq',
+            'general',
+            'legal_disclaimer',
+            'other',
+            'summary'
+        }
+
+    def _process_mandatory_and_optional(self, root_element):
+        notes = []
+
+        for elem_note in root_element.Note:
+
+            # mandatory
+            new_note = dict(
+                text=elem_note.text,
+                category=elem_note.get('Type'),
+            )
+
+            if new_note['category'].lower() not in self.enum_categories:
+                log_msg = f'Invalid document notes category ' \
+                          f'{new_note["category"]}. ' \
+                          f'Should be one of: {",".join(str(x) for x in sorted(self.enum_categories))}!'
+                if self.config.get('force', False):
+                    logging.warn(log_msg)
+                else:
+                    logging.error(log_msg)
+
+            # optional
+            if elem_note.get('Audience'):
+                new_note['audience'] = elem_note.get('Audience')
+            if elem_note.get('Title'):
+                new_note['title'] = elem_note.get('Title')
+
+            notes.append(new_note)
+
+        if len(notes) > 0:
```

Comment:
When notes is a list, a simple `if notes` Should do just fine. In case we allow `None` We are doomed with the `len` function (`TypeError`) but the simple condition nicely avoids that …

Reply:
nice to have as variables are pre-defined. But integrated.